src/bridge/transactions/signing_winternitz.rs

```diff
@@ -29,33 +29,58 @@ impl WinternitzSecret {
 }
 
 #[derive(Serialize, Deserialize, Eq, PartialEq, Hash, Clone)]
-pub struct WinternitzPublicKey(Vec<Vec<u8>>);
+pub struct WinternitzPublicKey<const TOTAL_DIGIT_COUNT: usize>(Vec<Vec<u8>>);
 
-impl From<&WinternitzSecret> for WinternitzPublicKey {
+impl<const TOTAL_DIGIT_COUNT: usize> From<&WinternitzSecret>
+    for WinternitzPublicKey<TOTAL_DIGIT_COUNT>
+{
     fn from(secret: &WinternitzSecret) -> Self {
         let mut public_key_vec = Vec::new();
-        for i in 0..N {
-            public_key_vec.push(public_key_for_digit(&secret.0, i).to_vec());
+        for i in 0..TOTAL_DIGIT_COUNT {
+            public_key_vec.push(public_key_for_digit(&secret.0, i.try_into().unwrap()).to_vec());
```

Comment:
Can we cast `i` to u32 instead of doing `try_into().unwrap()`?
```suggestion
        for i in 0..TOTAL_DIGIT_COUNT {
            public_key_vec.push(public_key_for_digit(&secret.0, i as u32).to_vec());
```

Reply:
We can. I just opted for a safer casting method ensuring no unintentional truncation takes place. While in practice, the values of `i` will never be large enough to cause truncation, this approach encourages caution.